content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,45 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+For every `dvc.yaml` file, a corresponding `dvc.lock` (YAML) file is created or
+updated by DVC as well. Its purpose is to lock the exact contents of all files
+involved in the pipeline(s) at the time that the pipeline was last committed
+(after using `dvc add`, `dvc run`, `dvc repro`, or `dvc commit`, for example).
+
+This means that regular <abbr>dependencies</abbr> and all <abbr>outputs</abbr>
+(including [metrics](/doc/command-reference/metrics) and
+[plots](/doc/command-reference/plots) files) are also listed in `dvc.lock`, but
+with additional fields that store the `md5` hash value (or `etag` for HTTP
+<abbr>external dependencies</abbr>) of each file or directory tracked by DVC.
+
+[parameter](/doc/command-reference/params#examples) key/value pairs are listed
+separately under `params`, grouped by parameters file. (No hash value is needed
+to lock params files, as these are not tracked as a whole.)
+
+Here's an example `dvc.lock` based on the one in [`dvc.yaml`](#dvcyaml-file)
+above:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
```

Comment:
Question @skshetry: Why do we need the `cmd` in dvc.lock? Thanks

Reply:
It is used to find out if `cmd` for stage changed, by comparing the `cmd` between `.lock` and `.yaml`. Kinda like hash that we had before.

Reply:
Makes sense. Info added.

Reply:
Thanks!